src/main/java/com/toofifty/easyblastfurnace/methods/GoldHybridMethod.java

```diff
@@ -77,34 +100,34 @@ public MethodStep next(BlastFurnaceState state)
             return waitForBars;
         }
 
-        if (state.getFurnace().has(barItem(), ItemID.GOLD_BAR)) {
-            if (!state.getEquipment().hasIceGlovesEffect()) {
-                return equipIceOrSmithsGloves;
-            }
-            return collectBars;
-        }
+		if (state.getConfig().tickPerfectMethod()) {
+			if (state.getFurnace().hasEnoughBars(barItem(), ItemID.GOLD_BAR)) {
+				if (!state.getEquipment().hasIceGlovesEffect()) {
+					return equipIceOrSmithsGloves;
+				}
+				return collectBars;
+			}
+		}
+		else if (state.getFurnace().has(barItem(), ItemID.GOLD_BAR)) {
+			if (!state.getEquipment().hasIceGlovesEffect()) {
+				return equipIceOrSmithsGloves;
+			}
+			return collectBars;
+		}
 
```

Comment:
I'm pretty sure that nothing inside of this `if (state.getBank().isOpen()) {` conditional ever runs because anything in a bank gets caught at the start now.

Reply:
Yes, I meant to go back and delete that. However I think the first if statement in that block needs to stay. That block checks for bars as well as ores, whereas the first `state.getBank().isOpen()` block only checks for ores.

Reply:
```JAVA
			if (state.getCoalBag().isFull() && state.getInventory().has(oreItem(), ItemID.GOLD_ORE)) {
				return putOntoConveyorBelt;
			}

			if (state.getInventory().has(ItemID.GOLD_BAR, barItem())) {
				return state.getConfig().useDepositInventory() ? depositInventory : depositBarsAndOres;
			}
```

These first two lines of the 1st bank check cover everything needed for gold ores and bars. Really, none of that code gets reached. A handy way to be sure is to run Runelite with the coverage option checked. The bits of code that actually get used have a green bar to the side. The bits that don't get used will be in red